contracts/smart-contract-wallet/test/SampleSocialRecoveryModule.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+import "../SmartAccount.sol";
+
+contract SocialRecoveryModule {
+    string public constant NAME = "Social Recovery Module";
+    string public constant VERSION = "0.1.0";
+    address public moduleOwner;
+
+    uint256 public threshold;
+    address[] public friends;
+
+    // isFriend mapping maps friend's address to friend status.
+    mapping(address => bool) public isFriend;
+    // isExecuted mapping maps data hash to execution status.
+    mapping(bytes32 => bool) public isExecuted;
+    // isConfirmed mapping maps data hash to friend's address to confirmation status.
+    mapping(bytes32 => mapping(address => bool)) public isConfirmed;
+
+    modifier onlyFriend() {
+        require(isFriend[msg.sender], "Method can only be called by a friend");
+        _;
+    }
+
+    function setup(
```

Comment:
this probably should not be a public method. anyone can be moduleOwner... what is the role of moduleOwner?